### PR TITLE
Fix cache bug. Allow search queries.

### PR DESCRIPTION
**Pull request by jerjou**
(as opposed to just wikipedia page titles)

### CODE REVIEW

File: app/wikipedia.py [hunk 1]
```diff
@@ -40,16 +40,30 @@ def get_article(titles=None, pageids=None):
...
         params['pageids'] = pageids
+    if query:
+        params['list'] = 'search'
+        params['srsearch'] = query
+
+        search_params = {
+            'list': 'search',
+            'srsearch': query
+        }
+        search_params.update(params)
```
Comment (tswast) [inline app/wikipedia.py:48]: ? wouldn't this just make search_params a copy of params?

Comment (jerjou) [reply app/wikipedia.py:48]: Oops. Lines 44 and 45 are vestigial.

There - does that make more sense?

File: app/main.py [hunk 2]
```diff
@@ -74,12 +74,16 @@ class ValidationError(Exception):
     pass
 
 
+INVALID_CACHE_KEY_CHARS = re.compile(r'\s')
```
Comment (theacodes) [inline app/main.py]: Constants always go at the top of the file.

Comment (theacodes) [inline app/main.py]: Actually this is trivial, don't bother making it a constant.

**Comment by theacodes** [comment]
LGTM

**Pull request merged**